lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsCallable.java

```diff
@@ -45,6 +45,9 @@
     protected final void assignArguments(List<?> args, Env e) {
         // TODO: var args
         for (int i=0; i< Math.min(args.size(),parameters.size()); i++) {
+            Class argClass = args.get(i) != null ? args.get(i).getClass() : Object.class;
```

Comment:
This is probably wrong. We do not know on a given invocation what the declared type is supposed to be. I think we need to use `Object` in all cases.